src/Stack/Options.hs

```diff
@@ -308,6 +312,25 @@ configOptsParser hide0 =
             hide
   where hide = hideMods hide0
 
+nixOptsParser :: Bool -> Parser NixOptsMonoid
+nixOptsParser showOptions =
+  NixOptsMonoid
+  <$> pure False
+  <*> maybeBoolFlags nixCmdName
+                     "using a Nix-shell"
+                     hide
+  <*> pure []
+  <*> pure Nothing
+  <*> many (textOption (long "nix-shell-options" <>
```

Comment:
Use `Options.Applicative.Args.argsOption` instead of `many . textOption`, which will handle options within the value properly.